### PR TITLE
docs: add `Response.json` static method

### PR DESCRIPTION
**Pull request by balazsorban44**
### Description

Newer runtimes implement this method and so it should be documented. 

### Motivation

It is part of the spec, see: https://fetch.spec.whatwg.org/#response-class

### Related issues and pull requests

https://github.com/whatwg/fetch/commit/b3bfd0c877fd09da8c315b43cf1dd858f3ac1612

https://github.com/whatwg/fetch/issues/1389

### CODE REVIEW

**Comment by sideshowbarker** [comment]
We can’t merge this patch as currently written, because the `{{domxref("Response.json()")}}` will create a link to the non-static method at https://developer.mozilla.org/en-US/docs/Web/API/Response/json — and we don’t yet have an article for the static method (see https://github.com/mdn/content/issues/21550), because we have a general issue that our title conventions don’t currently provide a way to have articles for the a static method and an instance method of the same object.

cc’ing @wbamberg for guidance on what to do here

**Comment by wbamberg** [comment]
Yeah, I saw this PR :(

There are 2 parts to enabling this:

1. retitling thousands of Web/API pages. I have been neglecting this but just filed a project at https://github.com/mdn/mdn/issues/284 which hopefully will enable us to work on it. I also just filed https://github.com/mdn/yari/pull/7825 which is a prerequisite for the retitling (it means we will still have sensible titles in sidebars and breadcrumbs).

2. have a convention for different URLs for static and non-static pages. I think we have a slightly fragile consensus on a `_static` prefix here. But this is dependent on (1) so I want to get (1) done first.

**Comment by sideshowbarker** [comment]
This PR doesn’t add `Response/json_static/index.md` article — instead it just adds a link and description to the `Response/index.md` article.

So it seems we still need for an `Response/json_static/index.md` article to be written before we can merge this.

File: files/en-us/web/api/response/index.md [hunk 1]
```diff
@@ -39,10 +39,12 @@ You can create a new `Response` object using the {{domxref("Response.Response",
...
 ## Static methods
 
-- {{domxref("Response.error()")}}
+- {{domxref("Response.error_static","Response.error()")}}
   - : Returns a new `Response` object associated with a network error.
-- {{domxref("Response.redirect()")}}
+- {{domxref("Response.redirect_static", "Response.redirect()")}}
   - : Creates a new response with a different URL.
+- {{domxref("Response.json_static", "Response.json()")}}
+  - : Creates a new response whose body is the JSON-encoded data, and status, status message, and headers are provided by init.
```
Comment (wbamberg) [inline files/en-us/web/api/response/index.md:47]:
```suggestion
  - : Creates a new response whose body is the provided JSON.
```

I think "and status, status message, and headers are provided by init" is a little hard to parse, and "init" isn't named like that in the linked page, and explaining it all here would start to be unwieldy for this entry, which is really just intended to be a link to the page and a very short summary (cf `redirect()` doesn't mention that you can set `status`, and that seems fine to me, too).

**Review by wbamberg — COMMENTED** [review]
Thank you! We can merge this, now we have https://github.com/mdn/content/pull/27154. I had a wording suggestion.

File: files/en-us/web/api/response/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/api/response/index.md:47]: @wbamberg Strongly agree. This method is a simple way to create JSON responses, so that's what I'll say. How it does so - i.e. in the body and by setting content type header is nice to have. But the point of these summaries is to know if you want to learn more.

So my variant, which I am going to accept and merge, is:

```suggestion
  - : Creates a new `Response` object for returning the provided JSON encoded data.
```

File: files/en-us/web/api/response/index.md [hunk 2]
```diff
@@ -39,10 +39,12 @@ You can create a new `Response` object using the {{domxref("Response.Response",
...
 ## Static methods
 
-- {{domxref("Response.error()")}}
+- {{domxref("Response.error_static","Response.error()")}}
   - : Returns a new `Response` object associated with a network error.
-- {{domxref("Response.redirect()")}}
+- {{domxref("Response.redirect_static", "Response.redirect()")}}
   - : Creates a new response with a different URL.
+- {{domxref("Response.json_static", "Response.json()")}}
+  - : Creates a new `Response` object for returning the provided JSON encoded data.
```
Comment (hamishwillee) [inline files/en-us/web/api/response/index.md:47]: Consistency change

```suggestion
- {{domxref("Response.redirect_static", "Response.redirect()")}}
  - : Returns a new response with a different URL.
- {{domxref("Response.json_static", "Response.json()")}}
  - : Returns a new `Response` object for returning the provided JSON encoded data.
```

**Review by hamishwillee — APPROVED** [review]
Merging once tests pass.  Thanks @balazsorban44 @wbamberg et al.

FYI, If it looks like I'm rushing this, it is because I need it for  #27169, and this is good now
